contents/blog/8020-the-data-warehouse.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "How we 80/20'ed the data warehouse"
+date: 2024-06-13
+author: 
+ - james-greenhill
+ - ian-vanagas
+featuredImage: https://res.cloudinary.com/dmukukwp6/image/upload/warehog_f318a079bf.png
+featuredImageType: full
+tags:
+  - Engineering
+  - Product engineers
+---
+
+If David Attenborough did documentaries about tech products, he would definitely do one about the mythical data warehouse. Because he doesn't, I'll take you on that adventure myself.
+
+We recently released our [own data warehouse](/docs/data-warehouse), but what does that mean? When other companies are building on top of, inside, and around data warehouses, why did we need to build our own and what's special about it?  
+
+This post dives into the strategic and technical details of why and how we built our version of the data warehouse.
+
+## PostHog vs the modern data stack
+
+The data warehouse space is a crowded one. They are the crown jewel of the [modern data stack](/blog/modern-data-stack-sucks), but we have a problem with this whole setup. The modern data stack is too complicated, requires too much maintenance, and creates a gap between engineers and the data they value.
+
+To improve this situation, PostHog aims to be the [source of truth for customer and product data](/handbook/why-does-posthog-exist#be-the-source-of-truth-for-customer-and-product-data). We also aim to be one of the first tools a startup installs. If we add more of the functionality startups and small teams need from the modern data stack, we can remove or delay their need for it.
+
+A big need is the flexibility a data warehouse provides. It enables them to query many types of data. The product data PostHog focuses on is one piece of that. We faced this challenge ourselves. We often need to query Stripe, Zendesk, and Hubspot data. A lot of it doesn't fit nicely into our event-based pattern, which is where building a data warehouse comes in.
```

Comment:
```suggestion
The data warehouse space is a crowded one. It's the crown jewel of the cursed [modern data stack](/blog/modern-data-stack-sucks) ecosystem, a concept so amorphous it's spawned an entire industry of consultants whose job is to define, sell and maintain the illusion it's working.

But (spoiler warning) we think the modern data stack kinda sucks for most companies. It's excessively complicated, hard to maintain, and widens the gap between engineers and the data they value. It doesn't add value, it extracts.

Our solution is to build more of the functionality startups and small teams need from the modern data stack, so we can remove or delay their need for it. We started by building a world-class analytics platform; the data warehouse is the next piece of that puzzle.

With it, our users can query many types of data from sources like Stripe, Zendesk and Hubspot – data that doesn't nicely fit into our existing event-based pattern. And they can do so without faffing about with middleware to get their data where they need it to be. 

In doing so, we can:

1. Be the source of truth for all product and customer data.
2. Be a default tool that every startup needs from the beginning.
```

Feel free to totally junk this. I was kind of going for something with more rhetorical heft, but maybe it's OTT? I like some of it, but it's a wordier.

Reply:
Bit of an aside, but I worry our language around warehouse being for startups and small teams could end up cementing the view we're "not for enterprise". Not super important here, but something for use to think about generally.